### DISCUSSION TITLE
Use comments to define message placeholder names

### DISCUSSION THREAD

**Discussion by AndrewIngram** [lingui/js-lingui#1982] [Ideas]
As we've been rolling out our use of Lingui, one thing that's particularly tedious is having to create a variable binding with a translator-friendly name just for the purposes of use in a string.

There've been a bunch of recent commits to Lingui that improve this in various ways, but I was musing on the possibility of some special comment syntax being supported by the macros to explicitly set the placeholder name, e.g something like:

```ts
t`Hello ${/* name */ users[0].firstName}`
```

I haven't thought through all the implications of this, so it might've already been considered and rejected.

**Comment by andrii-bodnar** [comment]
There was a similar discussion in #1874.

A great new feature was introduced in v5 that gives translators more context about the placeholders. You may find this useful. See #1965 for more details.

**Reply by AndrewIngram** [reply]
Checking back in here, we're now running on v5 with the new placeholder metadata enabled; but to be honest, it doesn't help much. We're finding the referenced variables very rarely do the trick in terms of providing a good explanation of what the placeholder means.

We get lots of things like this:
```
placeholder {0}: FORM_TAB_TO_LABEL[formTab]
```

It makes me feel like there's still a need to have a way to explicitly define what the placeholder shows up as in the translation string.

**Reply by timofei-iatsenko** [reply]
The latest proposition was here: https://github.com/lingui/js-lingui/pull/1874#issuecomment-1983664085
